### ISSUE TITLE
position of variants(output) is different from the vcf(input)

### ISSUE THREAD

**Issue by achilles1989** [10XGenomics/vartrix#27]
POS in the output variants = POS in the prepared vcf -1
I wonder how that happens and if it has an effect on the results？

**Comment by ifiddes-10x** [comment]
Ah, yes that is a (sort-of) bug. It won't have any effect on the results. Internally all of the work is being done on the 0-based coordinates, while VCF is 1-based. I assume you are talking about the output in `--out-variants`?

**Comment by achilles1989** [comment]
Thanks, you've solved my problem.

**Comment by Puriney** [comment]
I think it is better to make the formats consistent. The `--out-variants` result would confuse some downstream analysis, e.g., linking to the genes, searching against the known SNPs.

**Comment by jfx319** [comment]
I agree with @Puriney -- imo, if the input vcf is being passed with positions already chosen, the positions should also be written out that way, or the user will be surprised downstream due to the behavior of this `--out-variants` flag. Even the example illustrated in the main README.md on 10XGenomics/vartrix uses the first 2 columns of the vcf as input to Seurat, without modifying coordinates. And so, if the `--out-variants` is being provided as a convenience function, it should adhere to the same expectations. 

There's enough of a headache dealing with [0-based vs 1-based coordinate systems](https://www.biostars.org/p/84686/) already from the various file formats / genome hosts -- being surprised by this gotcha on top of that is no fun! 

From what I gather, `bcf::Reader` module is getting a record, which then calls its own pos() to get the position. According to the [rust docs](https://docs.rs/rust-htslib/0.29.0/src/rust_htslib/bcf/record.rs.html#163), this method is getting a 0-based position (not surprising for binary file formats like bcf). 

Therefore, changing [line 1175](https://github.com/10XGenomics/vartrix/blob/4948ba4a50a9f2cab3d27c3601e4646503132098/src/main.rs#L1175) should do it:
```rust
let pos = rec.pos() + 1;
```
not verified (I'm a bit rusty here...) pun intended
